shared/src/simulation/behaviors/assign-leader.ts

```diff
@@ -243,6 +239,29 @@ export const assignLeaderBehavior: SimulationBehavior<AssignLeaderBehaviorState>
         },
     };
 
+function selectNewLeader(
+    draftState: Mutable<ExerciseState>,
+    simulatedRegion: Mutable<SimulatedRegion>,
+    behaviorState: Mutable<AssignLeaderBehaviorState>
+) {
+    const personnel = Object.values(draftState.personnel).filter(
+        (pers) =>
+            isInSpecificSimulatedRegion(pers, simulatedRegion.id) &&
+            pers.personnelType !== 'notarzt'
+    );
+
+    if (personnel.length === 0) {
+        return;
+    }
```

Comment:
```suggestion
```
This would prevent a proper event if the leader is removed and there is no-one to replace them.

Reply:
The helper can now  change the leader even if there is no new leader so an event is sent